src/hooks/useFormWorkspaceSize.ts

```diff
@@ -0,0 +1,51 @@
+import { useLayoutEffect, useMemo, useState } from 'react';
+import { pxToRem } from '../utils/common-utils';
+
+/**
+ * The width of the supported workspace variants in rem
+ */
+const narrowWorkspaceWidth = 26.25;
+const widerWorkspaceWidth = 32.25;
+const extraWideWorkspaceWidth = 48.25;
+
+type WorkspaceSize = 'narrow' | 'wider' | 'extra-wide' | 'ultra-wide';
+
+/**
+ * This hook evaluates the size of the current workspace based on the width of the container element
+ */
+export function useFormWorkspaceSize(rootRef: React.RefObject<HTMLDivElement>): WorkspaceSize {
+  // width in rem
+  const [containerWidth, setContainerWidth] = useState(0);
+  const size = useMemo(() => {
+    if (containerWidth <= narrowWorkspaceWidth) {
+      return 'narrow';
+    } else if (containerWidth <= widerWorkspaceWidth) {
+      return 'wider';
+    } else if (containerWidth <= extraWideWorkspaceWidth) {
+      return 'extra-wide';
+    } else {
+      return 'ultra-wide';
+    }
+  }, [containerWidth]);
+
+  useLayoutEffect(() => {
+    const handleResize = () => {
+      const containerWidth = rootRef.current?.parentElement?.offsetWidth;
+      containerWidth && setContainerWidth(pxToRem(containerWidth));
+    };
+    handleResize();
+    const resizeObserver = new ResizeObserver((entries) => {
+      handleResize();
+    });
+
+    if (rootRef.current) {
```

Comment:
Shouldn't the check here actually be:
```suggestion
    if (rootRef.current?.parentElement) {
```